### PR TITLE
git: add pcre2 support

### PR DESCRIPTION
**Pull request by mimame**
###### Motivation for this change

Add pcre2 support to the last git version (2.14.0)

###### Things done

Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers.

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by mimadrid** [comment]
I don't know if enable pcre2 by default is a good a idea so don't hesitate to ask for changing it to be disabled by default

File: pkgs/applications/version-management/git-and-tools/git/default.nix [hunk 1]
```diff
@@ -70,7 +72,9 @@ stdenv.mkDerivation {
   # so that `SPARSE_FLAGS' corresponds to the current architecture...
   #doCheck = true;
 
-  installFlags = "NO_INSTALL_HARDLINKS=1";
+  installFlags = "NO_INSTALL_HARDLINKS=1"
+    + (if withpcre2 then " USE_LIBPCRE2=1" else "");
```
Comment (Mic92) [inline pkgs/applications/version-management/git-and-tools/git/default.nix]: Do you know, where git uses libpcre2? I assume it replaces a slower posix regex engine somewhere.

Comment (mimadrid) [inline pkgs/applications/version-management/git-and-tools/git/default.nix]: https://git-scm.com/docs/git-grep#git-grep--P

Comment (Mic92) [inline pkgs/applications/version-management/git-and-tools/git/default.nix]: Is `libpcre2` already part of the dependency closure? Then we can just enable it. An alternative would be to add it only to `gitFull`. I have no opinion, whether it should be also available in the `git` derivation.

Comment (vcunat) [inline pkgs/applications/version-management/git-and-tools/git/default.nix]: No, libpcre (1) is in the closure ATM.  Version 2 is rather new and rarely used so far.

Comment (vcunat) [inline pkgs/applications/version-management/git-and-tools/git/default.nix]: Still, pcre2 is (only) 1.5 MB and has no deps itself.

Comment (mimadrid) [inline pkgs/applications/version-management/git-and-tools/git/default.nix]: @vcunat only libpcre2 is recommended by git https://github.com/git/git/blob/master/Documentation/RelNotes/2.14.0.txt#L29
@Mic92 I think you're right and I've updated the pull request to disable pcre2 by default in gitMinimal

**Review by vcunat — APPROVED** [review]

**Pull request merged**